### DISCUSSION TITLE
Reducing brush size increases blockiness

### DISCUSSION THREAD

**Discussion by TokisanGames** [TokisanGames/Terrain3D#128] [Architecture]
Painting a 50m circle at 100% opacity is a bit blocky. But painting a 10m circle is worse! The edges should be 100% smooth if we're scaling down by 80%.

![image](https://github.com/outobugi/GDExtensionTerrain/assets/632766/2176d07e-e4e4-415d-bf69-f76076ecc652)

My painting approach. Search brush_size and mask_size.
https://github.com/TokisanGames/gterrain/blob/master/addons/gterrain/src/GTerrainPainter.gd

**Comment by outobugi** [comment]
Nope, nothing can be done about this. This is not a bug, it's how the shader works. The transition between textures is a blurred square (size of a map's pixel).

**Comment by TokisanGames** [comment]
Improvements to the painting tools and the shader have made this a non-issue. Archived.

**Comment by wenqiangwang** [comment]
We met with some issue with low poly textures (please see for details here: https://www.reddit.com/r/godot/comments/1j78q4d/unable_to_blend_synty_textures_with_terrain3d/)

Is that expected? Is there any technique to work it around?

![image](https://github.com/user-attachments/assets/adee5cef-5c95-45fe-9aa6-69f6250c1d32)

**Reply by TokisanGames** [reply]
I commented on reddit. That has nothing to do with this discussion. You need proper height textures and technique shown in tutorial video 2 and our docs.

**Reply by wenqiangwang** [reply]
That is very help @TokisanGames I added height map to the textures and the blocky went away. Thanks for the amazing plugin and your support!
